src/Sarif.Viewer.VisualStudio/Models/SarifErrorListItem.cs

```diff
@@ -490,6 +489,10 @@ public ResultTextMarker LineMarker
                 if (_lineMarker == null && Region != null && Region.StartLine > 0)
                 {
                     _lineMarker = new ResultTextMarker(_runId, Region, FileName);
+                    if (Locations.Count > 0)
+                    {
+                        _lineMarker.UriBaseId = Locations[0].UriBaseId;
```

Comment:
>UriBaseId [](start = 36, length = 9)

_This_ is the fix.

Reply:
Prefer:
```
_lineMarker = new ResultTextMarker(_runId, Region, FileName) {
    UriBaseId = Locations[0]?.UriBaseId,
};
```

Reply:
That doesn't actually work. `[0]?` doesn't mean "return `null` if the 0th element isn't there". It means "return `null` if `[0]` is `null`". If there is no 0th element, this throws "index out of range". I've made this mistake before :-)

---
In reply to: [498665970](https://github.com/microsoft/sarif-visualstudio-extension/pull/207#discussion_r498665970) [](ancestors = 498665970)

Reply:
Thanks. Obvious after you point it out. What are your thoughts on:
```
_lineMarker = new ResultTextMarker(_runId, Region, FileName) {
    UriBaseId = Locations.FirstOrDefault()?.UriBaseId,
};
```

Reply:
Sure, that's fine. It would have to be `Locations?.FirstOrDefault()?.UriBaseId` but yes.

---
In reply to: [499018240](https://github.com/microsoft/sarif-visualstudio-extension/pull/207#discussion_r499018240) [](ancestors = 499018240)

Reply:
Ah yes. Likewise the original code would have been `Locations?.Count` also right?

Reply:
LOL yes.

---
In reply to: [499028535](https://github.com/microsoft/sarif-visualstudio-extension/pull/207#discussion_r499028535) [](ancestors = 499028535)